src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -451,31 +450,46 @@ trait ClusterComponent extends LeoComponent {
         ClusterComponent.this.extensionQuery.unmarshallExtensions(userJupyterExtensionConfig),
         clusterRecord.autopauseThreshold,
         clusterRecord.defaultClientId,
-        clusterRecord.stopAfterCreation
+        clusterRecord.stopAfterCreation,
+        clusterImageRecords map ClusterComponent.this.clusterImageQuery.unmarshalClusterImage toSet
       )
     }
   }
 
-  // select * from cluster c left join label l on c.id = l.clusterId
-  val clusterQueryWithLabels: Query[(ClusterTable, Rep[Option[LabelTable]], Rep[Option[ExtensionTable]]), (ClusterRecord, Option[LabelRecord], Option[ExtensionRecord]), Seq] = {
+  // just clusters and labels: no instances, extensions, etc.
+  //   select * from cluster c
+  //   left join label l on c.id = l.clusterId
+  val minimalClusterQuery: Query[(ClusterTable, Rep[Option[LabelTable]]), (ClusterRecord, Option[LabelRecord]), Seq] = {
```

Comment:
I decided to rename these queries (and related `unmarshal*` methods) to `minimalClusterQuery` and `fullClusterQuery`. The former is used by "list clusters" and only joins clusters to labels. The latter is used by "get cluster" and does all the joins to retrieve cluster metadata.